apps/kyb-app/src/components/organisms/UIRenderer/elements/StepperUI/StepperUI.tsx

```diff
@@ -5,53 +5,54 @@ import { VerticalLayout } from '@/components/atoms/Stepper/layouts/Vertical';
 import { usePageResolverContext } from '@/components/organisms/DynamicUI/PageResolver/hooks/usePageResolverContext';
 import { useStateManagerContext } from '@/components/organisms/DynamicUI/StateManager/components/StateProvider';
 import { useDynamicUIContext } from '@/components/organisms/DynamicUI/hooks/useDynamicUIContext';
-import { useMemo, useRef } from 'react';
+import { useMemo, useState } from 'react';
 import { usePageContext } from '@/components/organisms/DynamicUI/Page';
 import { UIPage } from '@/domains/collection-flow';
-import { UIElementState } from '@/components/organisms/DynamicUI/hooks/useUIStateLogic/hooks/useUIElementsStateLogic/types';
 import { ErrorField } from '@/components/organisms/DynamicUI/rule-engines';
+import { CollectionFlowContext } from '@/domains/collection-flow/types/flow-context.types';
+import { isPageCompleted } from '@/helpers/prepareInitialUIState';
+import { UIElementState } from '@/components/organisms/DynamicUI/hooks/useUIStateLogic/hooks/useUIElementsStateLogic/types';
 
 export const StepperUI = () => {
   const { state: uiState } = useDynamicUIContext();
   const { pages, currentPage } = usePageResolverContext();
-  const { state } = useStateManagerContext();
+  const { state, payload } = useStateManagerContext();
   const { pageErrors } = usePageContext();
 
-  const initialPageNumber = useRef(currentPage?.number);
-
   const computeStepStatus = ({
-    uiElement,
-    page,
     pageError,
-    currentPage,
+    page,
+    context,
+    uiElementState,
   }: {
-    uiElement: UIElementState;
     page: UIPage;
+    uiElementState: UIElementState;
     pageError: Record<string, ErrorField>;
     currentPage: UIPage;
+    context: CollectionFlowContext;
   }) => {
-    if (!!Object.keys(pageError).length && currentPage.number === page.number) return 'warning';
     if (
-      uiElement?.isCompleted ||
-      page.number <=
-        // @ts-ignore
-        initialPageNumber?.current
+      Object.keys(pageError).length &&
+      Object.values(pageError).some(error => error.type === 'warning')
```

Comment:
If there are no keys there won't be any errors of type warning right? 🙂 
```suggestion
      Object.values(pageError ?? {}).some(error => error.type === 'warning')
```